tools/envoy/fetch.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+# This script fetches Envoy binary from download.konghq.com
+#
+# Requires:
+# - $BINARY_PATH, path where binary will be fetched, for example 'out/envoy'
+# - $ENVOY_DISTRO, name of the distributive (i.e darwin, linux)
+#
+# Optional:
+# - $ENVOY_TAG, git tag to reference specific revision
+# - $ENVOY_COMMIT_HASH, hash of the git commit. If specified, then $ENVOY_TAG will be ignored
+#
+# at least one of $ENVOY_TAG or $ENVOY_COMMIT_HASH should be specified
+
+set -e
+
+function msg_red() {
+  builtin echo -en "\033[1;31m" >&2
```

Comment:
could be worth having a `common` script that we source from our various build scripts to reduce noisiness of these scripts.

Reply:
done, but I'm not sure if this is a good way to import `common.sh` :D
```bash
source "$(dirname "$(dirname "$0")")/common.sh" # relative path to ../common.sh
```

Reply:
Maybe
```
source "$(dirname -- "${BASH_SOURCE[0]}")/../common.sh"
```
? I think `BASH_SOURCE[0]` works in case the script itself isn't used to invoke the script.

Reply:
Thank you! Updated the PR